core/trino-main/src/main/java/io/trino/server/remotetask/RemoteTaskCleaner.java

```diff
@@ -0,0 +1,116 @@
+package io.trino.server.remotetask;
+
+import com.google.errorprone.annotations.concurrent.GuardedBy;
+import io.airlift.http.client.HttpClient;
+import io.airlift.http.client.Request;
+import io.airlift.http.client.StatusResponseHandler;
+import io.airlift.log.Logger;
+import io.opentelemetry.api.trace.SpanBuilder;
+import io.trino.execution.TaskId;
+import io.trino.execution.TaskState;
+
+import java.net.URI;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.function.Supplier;
+
+import static io.airlift.http.client.HttpUriBuilder.uriBuilderFrom;
+import static io.airlift.http.client.Request.Builder.preparePost;
+import static java.util.Objects.requireNonNull;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+
+public class RemoteTaskCleaner
```

Comment:
Do we need a separate service for this? Cannot HttpRemoteTask do cleanup itself?

Reply:
It can - just having separate class is cleaner I think. I can put that as inner class too but I do not like that too much.